cmd/snap-confine/mount-support.c

```diff
@@ -140,6 +140,57 @@ static void setup_private_pts(void)
 	sc_do_mount("/dev/pts/ptmx", "/dev/ptmx", "none", MS_BIND, 0);
 }
 
+void sc_setup_snap_instance_mapping(const char *snap_instance)
+{
+	char snap_name[41] = { 0 };
```

Comment:
Looking at this I think we need to introduce libsnap-confine-private/limits.h with things like `SC_MAX_SNAP_NAME_LEN`. Otherwise this is just pure magic and bugs waiting to happen.